File: frontend/src/types/report.ts
export type ReportKey =
  | "attendances"
  | "deliveries"
  | "stock_movements"
  | "benefits"
  | "families"
  | "stock_alerts";

export interface ReportKpisResponse {
  families_served: number;
  baskets_delivered: number;
  items_distributed: number;
}

export interface ReportDefinitionResponse {
  key: ReportKey;
  title: string;
  description: string;
  format: "csv";
  uses_period: boolean;
}

export interface ReportsOverviewResponse {
  start_date: string;
  end_date: string;
  generated_at: string;
  kpis: ReportKpisResponse;
  available_reports: ReportDefinitionResponse[];
}

File: backend/app/schemas/report.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel


ReportKey = Literal[
    "attendances",
    "deliveries",
    "stock_movements",
    "benefits",
    "families",
    "stock_alerts",
]


class ReportKpisResponse(BaseModel):
    """Indicadores reais calculados para o período selecionado."""

    families_served: int
    baskets_delivered: int
    items_distributed: int


class ReportDefinitionResponse(BaseModel):
    """Relatório que o perfil autenticado pode gerar."""

    key: ReportKey
    title: str
    description: str
    format: Literal["csv"] = "csv"
    uses_period: bool = True


class ReportsOverviewResponse(BaseModel):
    """Visão consolidada e governada da área de relatórios."""

    start_date: date
    end_date: date
    generated_at: datetime
    kpis: ReportKpisResponse
    available_reports: list[ReportDefinitionResponse]
